ably.d.ts

```diff
@@ -2424,8 +2646,91 @@ export interface Message {
    * Allows a REST client to publish a message on behalf of a Realtime client. If you set this to the {@link Connection.key | private connection key} of a Realtime connection when publishing a message using a {@link RestClient}, the message will be published on behalf of that Realtime client. This property is only populated by a client performing a publish, and will never be populated on an inbound message.
    */
   connectionKey?: string;
+  /**
+   * A summary of all the annotations that have been made to the message. Will always be
+   * populated for a message.summary, and may be populated for any other type (in
+   * particular a message retrieved from REST history will have its latest summary
+   * included).
+   * The keys of the map are the annotation types. The exact structure of the value of
+   * each key depends on the aggregation part of the annotation type, e.g. for a type of
+   * reaction:distinct.v1, the value will be a DistinctValues object. New aggregation
+   * methods might be added serverside, hence the 'unknown' part of the sum type.
+   */
+  summary?: Map<string, SummaryEntry>;
```

Comment:
Should this be `Record<string, SummaryEntry>`? We don't actually create a Map, and maps are a bit awkward to work with in this case:

`summary.get(type)` instead of `summary[type]` and they'll need to be created all the time if parsing to/from JSON.

Reply:
Ah, yep, you're right, it should be Record. I would have expected typescript to catch that mistake, butit doesn't because implicit default index signatures mean that `summary['reaction:distinct.v1']` still typechecks and has a type of `any` :frowning_face: